### ISSUE TITLE
Complex signal required for SampledAcousticSource when using Propagation Models?

### ISSUE THREAD

**Issue by grahamrow** [org-arl/UnderwaterAcoustics.jl#20]
Hello and Happy New Year!

I was having fun with this package and wanted to try sending some more generic signals through the acoustic simulation API. No issues creating the Tx and looking at the signal.

```julia
env = UnderwaterEnvironment()
pm = PekerisRayModel(env,8)
rx = AcousticReceiver(500.0, -10.0)
tx = SampledAcousticSource(0, 0, -5.0, cos.(collect(1:1000)./10.0); fs=1000.0)
record(tx, 1.0, 1000.0)
```

When I try to run this with the propagation model things break.
```julia
record(pm, tx, rx, 1.0, 1000.0;)
```
```
InexactError: Float64(-0.0010541008282025442 + 0.000582555920034865im)
```

Looking through the source is seems like `Pinger`, etc., generate in-phase and quadrature signal components using `cis`. Indeed if I run 
```julia
tx = SampledAcousticSource(0, 0, -5.0, cis.(collect(1:1000)./10.0); fs=1000.0)
```
it works fine, as far as I can tell. Maybe some stronger typing is necessary here? It's also confusing to have the simpler command above work fine, though I understand why this is the case?

Thanks,
Graham

**Comment by mchitre** [comment]
The propagation modeling API is designed to work on complex analytic signals, as the phase of the signal needs to be carried through the propagation model (since reflecting boundaries or caustics can introduce phase changes). Hence you need the `cis`, rather than the `cos`. For a more general real passband signal `x`, you can simply do `analytic(x)` to get the complex passband signal.

We should improve the documentation to make it clearer.

**Comment by grahamrow** [comment]
Thanks for the clarification! Doc update would indeed be helpful, though you could still consider typing the signal as an `Array{Complex}` to prevent amateurs such as myself getting confused.

**Comment by mchitre** [comment]
Yes, thanks for the feedback. In fact, I will see if I can do a step better and automatically promote the array to an analytic version if you pass in a real one, and convert the output back to real in that case.
